.github/workflows/tests-ipsec-upgrade.yaml

```diff
@@ -122,14 +150,33 @@ jobs:
             SHA="${{ github.sha }}"
           fi
           echo sha=${SHA} >> $GITHUB_OUTPUT
-          CILIUM_DOWNGRADE_VERSION=$(contrib/scripts/print-downgrade-version.sh)
+          if [ "${{ matrix.mode }}" = "minor" ]; then
+            CILIUM_DOWNGRADE_VERSION=$(contrib/scripts/print-downgrade-version.sh)
+            IMAGE_TAG=${CILIUM_DOWNGRADE_VERSION}
+          else
+            # Upgrade from / downgrade to patch release.
+            # In some cases we expect to fail to get the version number, do not
+            # fail the workflow in such case. This includes:
+            # - on main branch where we don't have preceeding patch releases
+            # - on stable branches on top of release preparation commits, where
+            #   we bump the patch version number to 90 and we can't easily
```

Comment:
We don't bump the version number to 90 any more, but we use vX.Y-dev instead. There's still no dedicated patch release though so I guess this doesn't change anything.

Reply:
I didn't think we set the version in stable branch release prep commits at all :thinking:

Reply:
> We don't bump the version number to 90 any more, but we use vX.Y-dev instead. There's still no dedicated patch release though so I guess this doesn't change anything.

Correct, in this case this is the same thing as having `0` as the patch release for the script.

> I didn't think we set the version in stable branch release prep commits at all 🤔

Double-checking now, it seems you're correct here as well. I thought I saw it on stable branches, but I suppose this was just the successive preparation commits added to `main` at the time, in preparation for the new minor releases. It's no in use after we've forked a new stable branch, as far as I can see.

Reply:
What's the consequence of this? Does it mean that we should just drop these lines that incorrectly document the behaviour with stable branches, or is there also an impact on other parts of the workflow?

Reply:
There is no impact on other parts of the workflow. The script fails to retrieve the previous patch version when the VERSION file ends with 0 or 90.

- Ending with 90 does not in fact occur on stable branches, so that's just one fewer case to worry about. It also no longer occurs on main branch given that we're using another scheme; so it no longer occurs at all. We could remove the mentions to this case, and even remove this case from `print-dowgrade-version.sh`.
- What we have instead is the `x.y.0-dev` scheme, which for the script is equivalent to `x.y.0` and treated the same as `x.y.90` for finding the previous patch version: we fail to get it:

    ```console
    $ VERSION=1.15.0-dev ./contrib/scripts/print-downgrade-version.sh           
    v1.14
    $ VERSION=1.15.0-dev ./contrib/scripts/print-downgrade-version.sh patch
    ERROR: failed to deduce patch release previous to version '1.15.0-dev'
    ```

- The case where the patch release found in `VERSION` is 0 remains unchanged, this part of the comment is till valid; in this case we fail to retrieve the previous patch version, because we're most likely on `main` and we don't have a previous patch version to find.

So everything related to the version ending in `.90` can probably be removed, but in the meantime this is just dead code. The only case it would have an impact is if we reach `x.y.90` for real, in which case we'd skip the upgrade/dowgrade to/from `x.y.89`. Sounds like a minor concern at this time.

I can do a follow-up to clean up these bits but don't want to commit to it for this week.

Reply:
Awesome, thanks for the analysis. It would be great if you could follow up, but there's no rush.

Incidentally, there are some potentially relevant bash snippets around identifying the tags in a target branch here:

https://github.com/cilium/cilium/blob/3d3f69ce3386d836ea45219239f0d69daea7af7a/Documentation/check-crd-compat-table.sh#L44

If it makes sense, we could factor some of these out for reuse.

Reply:
OK, setting myself a reminder for the new year

Reply:
https://github.com/cilium/cilium/pull/30413